pkg/sources/source.go

```diff
@@ -117,7 +121,31 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 		return fmt.Errorf("unrecognized isb svc type %q", sp.ISBSvcType)
 	}
 
-	sourcer, err := sp.getSourcer(writers, fetchWatermark, publishWatermark, sourcePublisherStores, log)
+	t, err := transformer.NewGRPCBasedTransformer()
+	if err != nil {
+		return fmt.Errorf("failed to create gRPC client, %w", err)
+	}
+
+	if sp.VertexInstance.Vertex.HasUDTransformer() {
+		// Readiness check
+		if err = t.WaitUntilReady(ctx); err != nil {
+			return fmt.Errorf("failed on user defined transformer readiness check, %w", err)
+		}
+		defer func() {
+			err = t.CloseConn(ctx)
+			if err != nil {
+				log.Warnw("failed to close gRPC client conn", zap.Error(err))
```

Comment:
`log.Errorw`?

`Failed to xxx` - Capital the first letter in the log message.

Reply:
Both map and reduce udf are using `log.Warnw`, do we know why?